lms/static/scripts/frontend_apps/components/HideUntilLoad.tsx

```diff
@@ -0,0 +1,86 @@
+import { Card, CardContent, Spinner } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useMemo, useRef, useState } from 'preact/hooks';
+
+import type { ErrorLike } from '../errors';
+import { InitialLoadingContext } from '../utils/initial-loading-context';
+import ErrorDisplay from './ErrorDisplay';
+
+export type InitialLoadingProps = {
+  children: ComponentChildren;
+
+  /**
+   * CSS classes to apply to the top-level element, regardless of it being the
+   * loading indicator, error or children wrapper.
+   */
+  classes?: string | string[];
+
+  /**
+   * A component to render after initial load or when a fatal error has been
+   * reported.
+   * It is hidden during initial load.
+   */
+  footer?: ComponentChildren;
```

Comment:
> What are the reasons for treating the footer specially? I gather one of them is because it is rendered even if an error has happened, anything else?

Nop, just that. I explained the reasoning here https://github.com/hypothesis/lms/pull/6287#issuecomment-2154702140

The reason it is called `footer` is because it renders under the other elements.